sympy/matrices/common.py

```diff
@@ -1956,6 +1956,67 @@ def permute(self, perm, orientation='rows', direction='forward'):
         if orientation == 'cols':
             return self._eval_permute_cols(perm)
 
+#start
+
+ def rot90(self, k=1):
+        """Rotates Matrix by 90 degrees
+        Examples
+        ========
+        >>> from sympy import MatrixSymbol
+        >>> A = MatrixSymbol('A', 2, 2).as_explicit()
+        >>> A
+        Matrix([
+        [A[0, 0], A[0, 1]],
+        [A[1, 0], A[1, 1]]])
+        >>> A.rot90(k=0)
+        Matrix([
+        [A[0, 0], A[0, 1]],
+        [A[1, 0], A[1, 1]]])
+        >>> A.rot90(k=1)
+        Matrix([
+        [A[1, 0], A[0, 0]],
+        [A[1, 1], A[0, 1]]])
+        >>> A.rot90(k=-1)
+        Matrix([
+        [A[0, 1], A[1, 1]],
+        [A[0, 0], A[1, 0]]])
+        >>> A.rot90(k=2)
+        Matrix([
+        [A[1, 1], A[1, 0]],
+        [A[0, 1], A[0, 0]]])
+        >>> A.rot90(k=-2)
+        Matrix([
+        [A[1, 1], A[1, 0]],
+        [A[0, 1], A[0, 0]]])
+        >>> A.rot90(k=3)
+        Matrix([
+        [A[0, 1], A[1, 1]],
+        [A[0, 0], A[1, 0]]])
+        >>> A.rot90(k=-3)
+        Matrix([
+        [A[1, 0], A[0, 0]],
+        [A[1, 1], A[0, 1]]])
+        """
+
+        mod = k%4
+        if mod == 0:
+            return self
+        if mod == 1:
+            return self[::-1, ::].T
+        if mod == 2:
+            return self[::-1, ::-1].T
+        if mod == 3:
+            return self[::, ::-1].T
+        if mod == -1:
```

Comment:
If you take mod-4, these codes are not reachable

Reply:
Done sir.

Reply:
Actually, this was not the expected move. I was telling about removing `mod == -1`, `mod == -2` or such because of the redundancy.

Reply:
Okay sir. Shall I let mod +4 stay?